### ISSUE TITLE
Enhancement - Support for Pushover.net

### ISSUE THREAD

**Issue by scottbindas** [kjdElectronics/fd-tone-notify#6]
For those IOS users, PushBullet no longer offers an IOS App.  [Pushover.net](https://pushover.net/) supports both IOS/Google apps.  

It would be desirable that to also expose their "Message Priority" function so that for a larger county that has many FD's you can push FD Alerts for surrounding departments but allow them to be configured where a silent alert is all that is needed.

**Comment by Joey40** [comment]
Unfortunate that PushBullet decided to drop support for iOS and apparently never revisited this decision.  Would love to see support for an alternative.

**Comment by kdolan** [comment]
This is planned

**Comment by ComputerComa** [comment]
Just wanted to note I have kinda started working on this as it is a feature that I would like to see.
Progress [here](https://github.com/ComputerComa/fd-tone-notify)
I'm not sure if I am going about it correctly or not, so feel free to contact with suggestions or advice. 
Really love the project!
Thanks, ComputerComa

**Comment by kdolan** [comment]
Looks good so far! You will just need to finish implementing [notifiers](https://github.com/ComputerComa/fd-tone-notify/tree/daa2e3c7e22c5e771920a6cf9c6b9a80f4f18aff/notifiers)/index.js and add some tests and I would be happy to get that merged!

**Comment by ComputerComa** [comment]
Yeah, that was where I kinda hit a wall, I'm not sure I quite understand how to actually finish implementing it.

**Comment by kdolan** [comment]
@ComputerComa  You are SOO close. 

1. Add a method to https://github.com/ComputerComa/fd-tone-notify/blob/daa2e3c7e22c5e771920a6cf9c6b9a80f4f18aff/notifiers/index.js called `_pushoverNotifications(params, prePostType)`
2. Add method to https://github.com/ComputerComa/fd-tone-notify/blob/daa2e3c7e22c5e771920a6cf9c6b9a80f4f18aff/obj/NotificationParams.js  for getPushovers()
3. Finish implementing the `_pushoverNotifications(params, prePostType)` method. You should be able to pretty closely copy the pattern for PushBullet.
4. Test your code and see if it works!

**Comment by ComputerComa** [comment]
I think I'm pretty much finished. I'm not quite sure I have it implemented right though. Pushover registers as a notifier but never actually fires off. I did a test with the included tone examples and it catches it, (I edited the included example config to use pushover as the only notifier) but it just says in the debug logs that there are no PRE or POST notifications for pushover.

**Comment by kdolan** [comment]
@ComputerComa It looks like you just forgot to add the new methods to the pre/post notifications methods.
If you add [here](https://github.com/ComputerComa/fd-tone-notify/blob/072985897d7a9eaaf886b9e156df4e574cc235ce/notifiers/index.js#L29) and [here](https://github.com/ComputerComa/fd-tone-notify/blob/072985897d7a9eaaf886b9e156df4e574cc235ce/notifiers/index.js#L48) those notifications should start processing.

**Comment by OhioRadioScannerNetwork** [comment]
> I think I'm pretty much finished. I'm not quite sure I have it implemented right though. Pushover registers as a notifier but never actually fires off. I did a test with the included tone examples and it catches it, (I edited the included example config to use pushover as the only notifier) but it just says in the debug logs that there are no PRE or POST notifications for pushover.

Do you still have the code? I am trying to setup my whole county however I realized that pushbullet is not on iOS

**Comment by ComputerComa** [comment]
> > I think I'm pretty much finished. I'm not quite sure I have it implemented right though. Pushover registers as a notifier but never actually fires off. I did a test with the included tone examples and it catches it, (I edited the included example config to use pushover as the only notifier) but it just says in the debug logs that there are no PRE or POST notifications for pushover.
> 
> 
> 
> Do you still have the code? I am trying to setup my whole county however I realized that pushbullet is not on iOS

I unfortunately don't anymore. It was on an old laptop that I no longer have
